### PR TITLE
printjob command not working for windows

### PR DESCRIPTION
**Pull request by sakshamgarg**
#### Bug/feature Description
* *C:\Program Files (x86)\PBS>qsub -- pbs-sleep 1000
0.winbuild*
*C:\Program Files (x86)\PBS>printjob 0
Failed to connect to PBS dataservice
Could not resolve dataservice host winbuild*

#### Affected Platform(s)
* *Windows*

#### Cause / Analysis / Design
* *Bug: It gives a return code WSANOTINITIALISED (10093)*
* *The application is not able to access the database socket*

#### Solution Description
* *Calling WSAStartup and WSACleanup for initiating the use of Winsock DLL by the process*
* *Adding a batch script to choose between the two executable (printjob_svr & printjob_host) as per the requirement*

#### Testing logs/output
* *Running server and mom on single machine: [printjob_windows_log.txt](https://github.com/PBSPro/pbspro/files/2574723/printjob_windows_log.txt)*
* *Running server and mom separately: [printjob_windows_log_host.txt](https://github.com/PBSPro/pbspro/files/2574790/printjob_windows_log_host.txt)*

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by sakshamgarg** [comment]
@PBSPro/pbspro-maintainers kindly have a look in the PR.

File: src/tools/printjob.c [hunk 1]
```diff
@@ -556,6 +559,9 @@ char *argv[];
 				}
 				exit(1);
```
Comment (subhasisb) [inline src/tools/printjob.c]: Do we not need a winsock cleanup before this exit?

File: src/tools/printjob.c [hunk 2]
```diff
@@ -547,6 +547,9 @@ char *argv[];
 		if (fp < 0) {
 
 #ifdef PRINTJOBSVR
+#ifdef WIN32
```
Comment (subhasisb) [inline src/tools/printjob.c]: I think you need to move the winsock_init in the case of PRINTJOBSVR to before the loop (and so would the cleanup be after the loop finishes entirely, else we will do init and cleanup for every possible iteration.

**Comment by subhasisb** [comment]
LTGM  - please get another review and rebase and squash your commits

**Comment by sakshamgarg** [comment]
@subhasisb I have rebased and squashed my commits.

**Pull request merged**